generator/main.cpp

```diff
@@ -67,77 +68,162 @@ void displayHelp(GeneratorSet *generatorSet);
+      includes << QStringLiteral(".");
 
-      // Environment PYTHONQT_INCLUDE
-      QString includePath = getenv("PYTHONQT_INCLUDE");
-      if (!includePath.isEmpty())
-        includes += includePath.split(pathSplitter, Qt::SkipEmptyParts);
+      const QChar pathSplitter = QDir::listSeparator();
 
-      // Includes from the command line
-      if (!commandLineIncludes.isEmpty())
-        includes += commandLineIncludes.split(pathSplitter, Qt::SkipEmptyParts);
-      for (auto it = includes.begin(); it != includes.end();)
-      {
-        if (!QDir(*it).exists())
-        {
-          qWarning() << "Include path " << it->toUtf8() << " does not exist, ignoring it.";
-          it = includes.erase(it);
+      // From env var PYTHONQT_INCLUDE
+      const QString envInclude = qEnvironmentVariable("PYTHONQT_INCLUDE");
+      if (!envInclude.isEmpty()) {
+        QStringList envIncludes = envInclude.split(pathSplitter, Qt::SkipEmptyParts);
+        for(const QString& include: qAsConst(envIncludes)) {
+          if (isDir(include)) {
+            includes << include;
+          }
+          else {
+            qWarning() << "Include path" << include << "does not exist, ignoring.";
+          }
         }
-        else
-        {
-          ++it;
+      }
+
+      // CLI-provided include paths
+      if (!commandLineIncludes.isEmpty()) {
+        const QStringList cliIncludes = commandLineIncludes.split(QDir::listSeparator(), Qt::SkipEmptyParts);
+        for (const QString& include : cliIncludes) {
+          if (isDir(include)) {
+            includes << QDir::cleanPath(include);
+          }
+          else {
+            qWarning() << "Include path" << include << "does not exist, ignoring.";
+          }
         }
       }
 
-      // Include Qt
-      QString qtdir = getenv("QTDIR");
-      if (qtdir.isEmpty() || !QDir(qtdir).exists(qtdir))
-      {
-        QString reason = "The QTDIR environment variable " + qtdir.isEmpty() ?
-                         "is not set. " : "points to a non-existing directory. ";
-#if defined(Q_OS_MAC)
-        qWarning() << reason << "Assuming standard binary install using frameworks.";
-            QString frameworkDir = "/Library/Frameworks";
-            includes << (frameworkDir + "/QtXml.framework/Headers");
-            includes << (frameworkDir + "/QtNetwork.framework/Headers");
-            includes << (frameworkDir + "/QtCore.framework/Headers");
-            includes << (frameworkDir + "/QtGui.framework/Headers");
-            includes << (frameworkDir + "/QtOpenGL.framework/Headers");
-            includes << frameworkDir;
-#else
-        qWarning() << reason << "This may cause problems with finding the necessary include files.";
-#endif
+      // Prefer QLibraryInfo (works without QTDIR)
+      QString qtInclude = QLibraryInfo::location(QLibraryInfo::HeadersPath);
+      if (!isDir(qtInclude)) {
+        // Fallback to QTDIR/include
+        const QString qtDir = qEnvironmentVariable("QTDIR");
```

Comment:
I dislike a little bit the fact that now we can't generate the bindings for a different Qt version than the version with which the generator was built. Sure, this is a little bit of a corner case, but I know that I have used this at some time during development.

Could we get another override mechanism for the Qt include directory?

Reply:
> I dislike a little bit the fact that now we can't generate the bindings for a different Qt version than the version with which the generator was built. Sure, this is a little bit of a corner case, but I know that I have used this at some time during development.

For future reference, this has been addressed in https://github.com/MeVisLab/pythonqt/pull/313